keyboards/s7_elephant/keymaps/default/keymap.c

```diff
@@ -0,0 +1,95 @@
+),
+
+/* FN
+ * ,------.   ,-------------------------------------------------------------------------------------------------.
+ * |Reset |   |  F1   | F2  | F3  | F4  | F5  | F6  | F7  | F8  | F9  | F10 | F11 | F12 | PScr |  ScrLck  |Pause|
+ * |------|   |-------------------------------------------------------------------------------------------------|
+ * |      |   |        |     |     |     |     |     |     |     |     |     |     |     |     |          |     |
+ * |------|   |-------------------------------------------------------------------------------------------------|
+ * |      |   |         |Prev |Play |Next |     |     |     |     |VolD |Mute |VolU |     |               |PgUp |
+ * |------|   |-------------------------------------------------------------------------------------------------|
+ * |      |   |           |     |     |     |     |     |     |     |     |     |     |             |     |PgDn |
+ * |------|   |-------------------------------------------------------------------------------------------------|
+ * |EPRM  |   |      |       |       |                                   |         |           |     |    |     |
+ * `------'   `-------------------------------------------------------------------------------------------------'
+ */
+
+[_FN] = LAYOUT_ansi(
+RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_PSCR, KC_SLCK, KC_PAUS,
+_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+_______, _______, KC_MPRV, KC_MPLY, KC_MNXT, _______, _______, _______, _______, KC_VOLD, KC_MUTE, KC_VOLU, _______, _______,          KC_PGUP,
+_______, _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, KC_PGDN,
+EPRM,    _______, _______, _______,                        _______,                       _______, _______,          _______, _______, _______
+),
+};
+
+void caps_lock_indicator(void) {
+    // Turn LEDs On/Off for Caps Lock
+    if (host_keyboard_leds() & (1 << USB_LED_CAPS_LOCK)) {
+        rgblight_enable_noeeprom();
+    } else {
+        rgblight_disable_noeeprom();
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case EPRM:
+            if (record->event.pressed) {
+                // Reset to defaults
+                eeconfig_init();
+            }
+            return true;
+            break;
+        case KC_CAPS:
+            if (record->event.pressed) {
+                caps_lock_indicator();
+            }
+            return true;
+            break;
+    }
+    return true;
+}
+
+void matrix_scan_user(void) {
+     // Set the LEDs to not-too-bright white
+    rgblight_sethsv(0, 0, 80);
```

Comment:
You do not want this in the matrix scan.
Especially as this writes to eeprom. 

What you should do is is set `RGBLIGHT_LIMIT_VAL` to prevent it from exceeding the brightness.

Eg, add this to your config.h:
```c
#define RGBLIGHT_LIMIT_VAL 80
```

Reply:
In fact, it may be better to use this: 
https://docs.qmk.fm/#/custom_quantum_functions?id=led-control

Eg: 
```c
void led_set_user(uint8_t usb_led) {
    if (usb_led & (1<<USB_LED_CAPS_LOCK)) {
        rgblight_sethsv_noeeprom(0, 255, 255); // red for on
    } else {
        rgblight_sethsv_noeeprom(0, 0, 255); // white for off
    }
}
```

Reply:
You are right.

I had trouble taming the leds to my will and had many iterations of this code. Somehow this was left in. I've included the final version in the fix-up commit.

Reply:
awesome.  And I know exactly what you mean.  

.... I may have a reputation for my love of RGB. :D